azerothcore/azerothcore-wotlk
# chore(DB/gameobject): table `gameobject` cleanup

## Changes Proposed:
- Cleaning the `gameobject` table from duplicates using the formula for calculating the Euclidean distance (`d`) between two objects
  - `d` is less or equal to 0.1
  - `d` is bigger than 0.1 and less or equal to 0.5
- Removing duplicate minerals or herbs in realm
  - `d` is bigger than 0.5 and less or equal to 10.0
- Hopefully this can also increase performance or startup time
  - a total of 1,277 objects were removed

_THIS PR DOES NOT AFFECT GAME OBJECTS THAT ARE IN THE POOL_


## Issues Addressed:
- [My own suggestion](https://github.com/azerothcore/azerothcore-wotlk/issues/5379)


## Tests Performed:
- NOT TESTED!
- Hopefully, it will be through this PR that we can check it out :)


## How to Test the Changes:
- I don't know, seriously, because this fix removes above 1,200 duplicate game objects from realm
- Personally, I think this patch should be installed and tested for a long time, because there may be some side changes
- Of the things that can be checked fairly quickly are the removal of duplicate herbs or minerals. Just look at the GUID that have been removed and what original GUIDs they refer to. Make a `.go` and check that there is no longer a duplicate there


## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
